CI tests for `vmware_guest_find` are failing. Is this related to the changes in 

```
2017-06-26 04:13:55 TASK [vmware_guest_find : find folders for each vm] ****************************
2017-06-26 04:13:55 task path: /root/ansible/test/integration/targets/vmware_guest_find/tasks/main.yml:35
...
2017-06-26 04:13:58 Using module file /root/ansible/lib/ansible/modules/cloud/vmware/vmware_guest_find.py
2017-06-26 04:13:58 <127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
2017-06-26 04:13:58 <127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392 `" && echo ansible-tmp-1498450438.63-139967742216392="` echo /root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392 `" ) && sleep 0'
2017-06-26 04:13:58 <127.0.0.1> PUT /tmp/tmpAqL9Qf TO /root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392/vmware_guest_find.py
2017-06-26 04:13:58 <127.0.0.1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392/ /root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392/vmware_guest_find.py && sleep 0'
2017-06-26 04:13:58 <127.0.0.1> EXEC /bin/sh -c '/tmp/ansible-test-coverage-ntvRjl/coverage/injector.py /root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392/vmware_guest_find.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392/" > /dev/null 2>&1 && sleep 0'
2017-06-26 04:13:58 The full traceback is:
2017-06-26 04:13:58 Traceback (most recent call last):
2017-06-26 04:13:58   File "/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py", line 281, in <module>
2017-06-26 04:13:58     main()
2017-06-26 04:13:58   File "/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py", line 267, in main
2017-06-26 04:13:58     uuid=module.params['uuid']
2017-06-26 04:13:58   File "/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py", line 110, in getvm_folder_paths
2017-06-26 04:13:58     self.getfolders()
2017-06-26 04:13:58   File "/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py", line 191, in getfolders
2017-06-26 04:13:58     self.folders = self._build_folder_tree(self.datacenter.vmFolder)
2017-06-26 04:13:58 AttributeError: 'NoneType' object has no attribute 'vmFolder'
2017-06-26 04:13:58 
2017-06-26 04:13:58 failed: [testhost] (item=/F0/DC0/network/VM Network) => {
2017-06-26 04:13:58     "changed": false, 
2017-06-26 04:13:58     "failed": true, 
2017-06-26 04:13:58     "item": "/F0/DC0/network/VM Network", 
2017-06-26 04:13:58     "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py\", line 281, in <module>\n    main()\n  File \"/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py\", line 267, in main\n    uuid=module.params['uuid']\n  File \"/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py\", line 110, in getvm_folder_paths\n    self.getfolders()\n  File \"/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py\", line 191, in getfolders\n    self.folders = self._build_folder_tree(self.datacenter.vmFolder)\nAttributeError: 'NoneType' object has no attribute 'vmFolder'\n", 
2017-06-26 04:13:58     "module_stdout": "", 
2017-06-26 04:13:58     "msg": "MODULE FAILURE", 
2017-06-26 04:13:58     "rc": 0
2017-06-26 04:13:58 }

```

--- a/lib/ansible/modules/cloud/vmware/vmware_vswitch.py
+++ b/lib/ansible/modules/cloud/vmware/vmware_vswitch.py
@@ -91,10 +91,12 @@ try:
     HAS_PYVMOMI = True
 except ImportError:
     HAS_PYVMOMI = False
+from ansible.module_utils.vmware import vmware_argument_spec, get_all_objs, connect_to_api
+from ansible.module_utils.basic import AnsibleModule
 
 
 def find_vswitch_by_name(host, vswitch_name):
-    for vss in host.config.network.vswitch:
+    for vss in host.configManager.networkSystem.networkInfo.vswitch:
         if vss.name == vswitch_name:
             return vss
     return None
@@ -136,7 +138,6 @@ class VMwareHostVirtualSwitch(object):
             self.module.fail_json(msg=method_fault.msg)
         except Exception as e:
             self.module.fail_json(msg=str(e))
-
 
     # Source from
     # https://github.com/rreubenur/pyvmomi-community-samples/blob/patch-1/samples/create_vswitch.py
@@ -180,7 +181,7 @@ class VMwareHostVirtualSwitch(object):
         if not host:
             self.module.fail_json(msg="Unable to find host")
 
-        self.host_system = host.keys()[0]
+        self.host_system = list(host.keys())[0]
         self.vss = find_vswitch_by_name(self.host_system, self.switch_name)
 
         if self.vss is None:
@@ -192,10 +193,10 @@ class VMwareHostVirtualSwitch(object):
 def main():
     argument_spec = vmware_argument_spec()
     argument_spec.update(dict(switch_name=dict(required=True, type='str'),
-                         nic_name=dict(required=False, type='str'),
-                         number_of_ports=dict(required=False, type='int', default=128),
-                         mtu=dict(required=False, type='int', default=1500),
-                         state=dict(default='present', choices=['present', 'absent'], type='str')))
+                              nic_name=dict(required=False, type='str'),
+                              number_of_ports=dict(required=False, type='int', default=128),
+                              mtu=dict(required=False, type='int', default=1500),
+                              state=dict(default='present', choices=['present', 'absent'], type='str')))
 
     module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
 
@@ -204,9 +205,6 @@ def main():
 
     host_virtual_switch = VMwareHostVirtualSwitch(module)
     host_virtual_switch.process_state()
-
-from ansible.module_utils.vmware import *
-from ansible.module_utils.basic import *
 
 if __name__ == '__main__':
     main()

--- a/test/integration/targets/vmware_guest_find/tasks/main.yml
+++ b/test/integration/targets/vmware_guest_find/tasks/main.yml
@@ -1,7 +1,7 @@
 - name: make sure pyvmomi is installed
   pip:
     name: pyvmomi
-    state: latest    
+    state: latest
 
 - name: store the vcenter container ip
   set_fact:

--- a/test/integration/targets/vmware_vswitch/aliases
+++ b/test/integration/targets/vmware_vswitch/aliases
@@ -1,0 +1,2 @@
+posix/ci/cloud/vcenter
+cloud/vcenter

--- a/test/integration/targets/vmware_vswitch/tasks/main.yml
+++ b/test/integration/targets/vmware_vswitch/tasks/main.yml
@@ -1,0 +1,57 @@
+# Test code for the vmware_vswitch
+# (c) 2017, Abhijeet Kasurde <akasurde@redhat.com>
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+- name: make sure pyvmomi is installed
+  pip:
+    name: pyvmomi
+    state: latest
+
+- name: store the vcenter container ip
+  set_fact:
+    vcsim: "{{ lookup('env', 'vcenter_host') }}"
+- debug: var=vcsim
+
+- name: kill vcsim
+  uri:
+    url: "{{ 'http://' + vcsim + ':5000/killall' }}"
+- name: start vcsim
+  uri:
+    url: "{{ 'http://' + vcsim + ':5000/spawn?cluster=2' }}"
+  register: vcsim_instance
+
+- debug: var=vcsim_instance
+
+- name: find folders for each vm
+  vmware_vswitch:
+    validate_certs: False
+    hostname: "{{ vcsim }}"
+    username: "{{ vcsim_instance['json']['username'] }}"
+    password: "{{ vcsim_instance['json']['password'] }}"
+    switch_name: vmswitch_0001
+    nic_name: vnic_1
+  register: vswitch
+
+- debug: var=vswitch
+
+# Currently, we don't return anything about vswitch, will
+# extend this TC once we have something to test more than
+# the state of task.
+- assert:
+    that:
+        - "{{ vswitch.changed == true }}"

--- a/test/sanity/pep8/legacy-files.txt
+++ b/test/sanity/pep8/legacy-files.txt
@@ -203,7 +203,6 @@ lib/ansible/modules/cloud/vmware/vmware_migrate_vmk.py
 lib/ansible/modules/cloud/vmware/vmware_target_canonical_facts.py
 lib/ansible/modules/cloud/vmware/vmware_vm_shell.py
 lib/ansible/modules/cloud/vmware/vmware_vmotion.py
-lib/ansible/modules/cloud/vmware/vmware_vswitch.py
 lib/ansible/modules/cloud/vmware/vsphere_copy.py
 lib/ansible/modules/cloud/vmware/vsphere_guest.py
 lib/ansible/modules/cloud/webfaction/webfaction_app.py

